test/tests/functional/pbs_checkpoint.py

```diff
@@ -0,0 +1,94 @@
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestCheckpoint(TestFunctional):
+    """
+    This test suite targets Checkpoint functionality.
+    """
+
+    def num_compute_vnodes(self):
+        """
+        Count the Cray compute nodes and return the value.
+        """
+        vlist = []
+        vnl = self.server.filter(
+            VNODE, {'resources_available.vntype': 'cray_compute'})
+        vlist = vnl["resources_available.vntype=cray_compute"]
+        nv = len(vlist)
+        self.assertNotEqual(nv, 0, "There are no cray_compute vnodes present.")
+        return nv
+
+    def test_checkpoint_abort_with_preemept(self):
+        abort_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        abort_file = self.du.create_temp_file(body=abort_script)
+        self.du.chmod(path=abort_file, mode=0755)
+        c = {'$action': 'checkpoint_abort 30 !' + abort_file + ' %sid'}
+        self.mom.add_config(c)
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+        self.platform = DshUtils().get_platform()
+        if self.platform != 'cray' and self.platform != 'craysim':
+            node_attrs = self.server.status(NODE, id=self.server.shortname)[0]
+            ncpus = node_attrs['resources_available.ncpus']
+            attrs = {'Resource_List.select': '1:ncpus=' + ncpus}
+        else:
+            nv = self.num_compute_vnodes()
+            attrs = {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                     ATTR_l + '.place': 'scatter'}
+
+        j1 = Job(TEST_USER, attrs)
+        j1.set_sleep_time(5)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        attrs['queue'] = 'expressq'
+        j2 = Job(TEST_USER, attrs)
+        j2.set_sleep_time(1)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        self.server.delete(id=jid1)
```

Comment:
Since you're giving the jobs such short sleep times, can't you let both jobs finish normally?  That way we should see j2 end and j1 run again and then also end without errors.

Reply:
that would be testing if the restart works well and according to me that is out of scope of this bug.

Reply:
when I checked, the job was not deleted.

Reply:
That is because you only have 1 test.  Jobs get deleted in setUp() of the following test.